src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -854,40 +732,28 @@ private void codeUnwrapReturnValue(MethodVisitor mv, Class<?> type) {
          * the Method object for this proxy method. A class loader is
          * anticipated at local variable index 0.
          */
-        private void codeFieldInitialization(MethodVisitor mv, String className) {
-            codeClassForName(mv, fromClass);
-
-            mv.visitLdcInsn(method.getName());
-
-            emitIconstInsn(mv, parameterTypes.length);
+        private void codeFieldInitialization(CodeBuilder cob, ClassDesc className) {
+            codeClassForName(cob, fromClass);
 
-            mv.visitTypeInsn(Opcodes.ANEWARRAY, JL_CLASS);
+            cob.ldc(method.getName())
+               .constantInstruction(parameterTypes.length)
+               .anewarray(CD_Class);
 
             // Construct an array with the parameter types mapping primitives to Wrapper types
             for (int i = 0; i < parameterTypes.length; i++) {
-                mv.visitInsn(DUP);
-                emitIconstInsn(mv, i);
-
+                cob.dup()
+                   .constantInstruction(i);
                 if (parameterTypes[i].isPrimitive()) {
-                    PrimitiveTypeInfo prim =
-                            PrimitiveTypeInfo.get(parameterTypes[i]);
-                    mv.visitFieldInsn(GETSTATIC,
-                            prim.wrapperClassName, "TYPE", LJL_CLASS);
+                    PrimitiveTypeInfo prim = PrimitiveTypeInfo.get(parameterTypes[i]);
+                    cob.getstatic(prim.wrapperClass, "TYPE", CD_Class);
```

Comment:
These can use a condy via [`ConstantBootstraps​::primitiveClass​(MethodHandles.Lookup, String, Class<?>)`].

[`ConstantBootstraps​::primitiveClass​(MethodHandles.Lookup, String, Class<?>)`]: https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/invoke/ConstantBootstraps.html#primitiveClass(java.lang.invoke.MethodHandles.Lookup,java.lang.String,java.lang.Class)

Reply:
I recommend optimizing in another patch, as this patch aims to preserve code parity.